crates/router/src/connector/payu/transformers.rs

```diff
@@ -199,7 +199,9 @@ impl<F, T>
                 mandate_reference: None,
                 connector_metadata: None,
                 network_txn_id: None,
-                connector_response_reference_id: None,
+                connector_response_reference_id: Some(
+                    item.response.ext_order_id.unwrap_or(item.response.order_id)
+                ),
```

Comment:
```suggestion
                connector_response_reference_id: item.response.ext_order_id.or(Some(item.response.order_id)),
```
Hello @harshalkh ,

You are unwrapping and then if the value is Some we are wrapping it again, whereas we can use or to reduce that one redundant operation . Please change it in other places as well .